README.md

```diff
@@ -1,51 +1,161 @@
+1. Change the current directory to where the project was cloned.
+
+## Building and running
+
+1. Connect a USB cable between the USB port on the target and the host computer.
+1. Run the following command to build the example project and program the microcontroller flash memory:
+
+   ```bash
+   $ mbed compile -m <TARGET> -t <TOOLCHAIN> --flash --sterm
+   ```
+   
+Note: You can use the Mbed CLI command-line option "--sterm" to open a serial terminal after flashing.
+
+Your PC may take a few minutes to compile your code.
+
+The binary is located at `./BUILD/<TARGET>/<TOOLCHAIN>/mbed-os-example-blinky-baremetal.bin`.
+
+Alternatively, you can manually copy the binary to the target, which gets mounted on the host computer through USB.
+
+## Expected output 
+
+The LED on your target turns on and off every 500 milliseconds and the serial terminal shows an output similar to: 
+
+``` 
+--- Terminal on /dev/tty.usbmodem21102 - 9600,8,N,1 ---
+This is the bare metal blinky example running on Mbed OS 99.99.99.
+``` 
+
+## Configuring the application
+
+### The bare metal mode
+
+The bare metal mode is a configuration of Mbed OS that excludes the RTOS as well as other features. It is specifically targeted for memory constrained devices because it gives developers higher degree of control of their system. See more details [here](https://os.mbed.com/docs/mbed-os/v5.15/reference/mbed-os-bare-metal.html)
+
+To build with the bare metal profile, the application configuration file must contain the following:
 ```
-mbed import mbed-os-example-blinky-baremetal
-cd mbed-os-example-blinky-baremetal
+{
+    "requires": ["bare-metal"]
+}
 ```
 
-### Now compile
+Note: For convenience, a few configuration files are present in the `configs` directory so you can try some of the different configurations shown in this example.
+
+### Futher optimisations
+
+#### Linking with smaller C libraries
+
+Both the `ARM` and `GCC_ARM` toolchains support optimised versions of their C standard libraries, microlib and newlib-nano respectively and we recommend using them with the bare metal profile.
+
+To build with the smaller C libraries, the application configuration file needs to be modified as follows:
+```
+{
+    "target_overrides": {
+        "*": {
+            "target.c_lib": "small"
+        }
+    }
+}
+```
+#### Using Mbed minimal printf library
 
-Invoke `mbed compile`, and specify the name of your platform and your favorite toolchain (`GCC_ARM`, `ARM`, `IAR`). For example, for the ARM Compiler:
+Mbed OS offers a smaller printf() alternative. The [minimal printf](https://github.com/ARMmbed/mbed-os/blob/master/platform/source/minimal-printf/README.md) library implements a subset of the `v/s/f/printf` function family and floating points can be disabled to save additional code size.
 
+To build with the minimal printf library and disable floating points printing, the application configuration file needs to be modified as follows:
```

Comment:
We can add a note that further optimisations are possible and details are in README.